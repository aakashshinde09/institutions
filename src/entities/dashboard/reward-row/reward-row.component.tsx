import type { RewardLookup } from '@api/data-contracts'
import { Tooltip } from '@mantine/core'
import { useGetNgoListQuery } from '@shared/api'
import { cn, formatDate, formatNumber, getFullNumber } from '@shared/lib/utils'
import { useMemo } from 'react'

interface IRewardRowProperties {
  data: RewardLookup
  className?: string
  index: number
}

export const RewardRow = ({ data, className, index }: IRewardRowProperties) => {
  const { data: allNgos } = useGetNgoListQuery()

  const currentNgo = useMemo(() => {
    const ngo = allNgos?.data?.items?.find((x) => x.ngoAddress === data.ngoAddress)
    return ngo ?? { name: 'Unknown', imageLink: '' }
  }, [allNgos?.data?.items, data.ngoAddress])

  return (
    <tr className={cn('tableRow', className)}>
      <td>#{index}</td>
      <td className="truncate text-center">
        {data.stake ? data.stake.customStakeId : 0}
      </td>

      <Tooltip label={formatNumber(data.balanceWithoutProfit)}>
        <td>{formatNumber(data.balanceWithoutProfit)} stETH</td>
      </Tooltip>
      <Tooltip label={getFullNumber(data.profit)}>
        <td className="truncate" title={`${data.profit} stETH`}>
          {formatNumber(data.profit, true)} stETH
        </td>
      </Tooltip>
      <td>
        <div className="flex items-center gap-3" title={currentNgo.name}>
          <img
            src={currentNgo.imageLink}
            alt={currentNgo.name}
            className="h-6 w-6 rounded-full"
          />
          <span className="max-w-[15rem] truncate">{currentNgo.name}</span>
        </div>
      </td>
      <Tooltip label={getFullNumber(data.donation)}>
        <td className="truncate" title={`${data.donation} stETH`}>
          {formatNumber(data.donation, true)} stETH
        </td>
      </Tooltip>

      <Tooltip label={getFullNumber(data.serviceFee)}>
        <td className="truncate" title={`${data.serviceFee} stETH`}>
          {formatNumber(data.serviceFee, true)} stETH
        </td>
      </Tooltip>
      <td className="text-muted">{formatDate(data.createdAt)}</td>
    </tr>
  )
}
