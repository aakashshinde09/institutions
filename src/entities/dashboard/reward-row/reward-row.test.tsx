import { expect, test, describe } from 'vitest'
import { render } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { RewardRow } from './reward-row.component'

const mock = {
  balanceWithoutProfit: 0.01,
  createdAt: '2024-05-05T19:28:02.000Z',
  donation: 0.000000049,
  id: 241,
  ngoAddress: '0xa72afa79826df92b88831b9ebd1a459a75345231',
  profit: 0.00000019,
  serviceFee: 1.299552152886571e-9,
  stackedAmount: 0.01,
}

describe('<RewardRow/>', () => {
  test('Renders correctly', () => {
    const component = render(
      <QueryClientProvider client={new QueryClient()}>
        <RewardRow data={mock} index={1} />
      </QueryClientProvider>,
    )

    expect(component).toMatchSnapshot()
  })
})
