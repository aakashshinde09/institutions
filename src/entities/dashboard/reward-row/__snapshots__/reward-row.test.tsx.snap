// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<RewardRow/> > Renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr
        class="tableRow"
      >
        <td>
          #
          1
        </td>
        <td
          class="truncate text-center"
        >
          0
        </td>
        <td>
          0.01
           stETH
        </td>
        <td
          class="truncate"
          title="1.9e-7 stETH"
        >
          0.00000019
           stETH
        </td>
        <td>
          <div
            class="flex items-center gap-3"
            title="Unknown"
          >
            <img
              alt="Unknown"
              class="h-6 w-6 rounded-full"
              src=""
            />
            <span
              class="max-w-[15rem] truncate"
            >
              Unknown
            </span>
          </div>
        </td>
        <td
          class="truncate"
          title="4.9e-8 stETH"
        >
          0.000000049
           stETH
        </td>
        <td
          class="truncate"
          title="1.299552152886571e-9 stETH"
        >
          0.0000000013
           stETH
        </td>
        <td
          class="text-muted"
        >
          05/05/2024
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr
      class="tableRow"
    >
      <td>
        #
        1
      </td>
      <td
        class="truncate text-center"
      >
        0
      </td>
      <td>
        0.01
         stETH
      </td>
      <td
        class="truncate"
        title="1.9e-7 stETH"
      >
        0.00000019
         stETH
      </td>
      <td>
        <div
          class="flex items-center gap-3"
          title="Unknown"
        >
          <img
            alt="Unknown"
            class="h-6 w-6 rounded-full"
            src=""
          />
          <span
            class="max-w-[15rem] truncate"
          >
            Unknown
          </span>
        </div>
      </td>
      <td
        class="truncate"
        title="4.9e-8 stETH"
      >
        0.000000049
         stETH
      </td>
      <td
        class="truncate"
        title="1.299552152886571e-9 stETH"
      >
        0.0000000013
         stETH
      </td>
      <td
        class="text-muted"
      >
        05/05/2024
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
